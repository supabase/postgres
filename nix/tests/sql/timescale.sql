-- Create schema v
create schema v;

-- Create a table in the v schema
create table v.sensor_data (
  time timestamptz not null,
  sensor_id int not null,
  temperature double precision not null,
  humidity double precision not null
);

-- Convert the table to a hypertable
select create_hypertable('v.sensor_data', 'time');

-- Insert some data into the hypertable
insert into v.sensor_data (time, sensor_id, temperature, humidity)
values 
  ('2024-08-09', 1, 22.5, 60.2),
  ('2024-08-08', 1, 23.0, 59.1),
  ('2024-08-07', 2, 21.7, 63.3);

-- Select data from the hypertable
select
  *
from
  v.sensor_data;

-- Drop schema v and all its entities
drop schema v cascade;

