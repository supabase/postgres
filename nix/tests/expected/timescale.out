-- Create schema v
create schema v;
-- Create a table in the v schema
create table v.sensor_data (
  time timestamptz not null,
  sensor_id int not null,
  temperature double precision not null,
  humidity double precision not null
);
-- Convert the table to a hypertable
select create_hypertable('v.sensor_data', 'time');
  create_hypertable  
---------------------
 (1,v,sensor_data,t)
(1 row)

-- Insert some data into the hypertable
insert into v.sensor_data (time, sensor_id, temperature, humidity)
values 
  ('2024-08-09', 1, 22.5, 60.2),
  ('2024-08-08', 1, 23.0, 59.1),
  ('2024-08-07', 2, 21.7, 63.3);
-- Select data from the hypertable
select
  *
from
  v.sensor_data;
             time             | sensor_id | temperature | humidity 
------------------------------+-----------+-------------+----------
 Fri Aug 09 00:00:00 2024 PDT |         1 |        22.5 |     60.2
 Thu Aug 08 00:00:00 2024 PDT |         1 |          23 |     59.1
 Wed Aug 07 00:00:00 2024 PDT |         2 |        21.7 |     63.3
(3 rows)

-- Drop schema v and all its entities
drop schema v cascade;
NOTICE:  drop cascades to 3 other objects
DETAIL:  drop cascades to table v.sensor_data
drop cascades to table _timescaledb_internal._hyper_1_1_chunk
drop cascades to table _timescaledb_internal._hyper_1_2_chunk
